sn_client/src/audit/spend_dag_building.rs

```diff
@@ -17,14 +17,18 @@ impl Client {
     /// Builds a SpendDag from a given SpendAddress recursively following descendants all the way to UTxOs
     /// Started from Genesis this gives the entire SpendDag of the Network at a certain point in time
     /// Does not verify the validity of the transactions
-    pub async fn build_spend_dag_from(&self, spend_addr: SpendAddress) -> WalletResult<SpendDag> {
+    pub async fn spend_dag_build_from(&self, spend_addr: SpendAddress) -> WalletResult<SpendDag> {
         let mut dag = SpendDag::new();
 
         // get first spend
-        let first_spend = self
-            .get_spend_from_network(spend_addr)
-            .await
-            .map_err(|err| WalletError::CouldNotVerifyTransfer(err.to_string()))?;
+        let first_spend = match self.get_spend_from_network(spend_addr).await {
+            Ok(s) => s,
+            Err(Error::MissingSpendRecord(_)) => {
+                trace!("UTXO at {spend_addr:?}");
+                return Ok(dag);
```

Comment:
1, Why it's not an `error` to be returned if spend not found ?
2, The comment of the function doesn't explain this situation

if it's a value return, maybe put an explaination in the function comment ?

Reply:
If a Spend is not found, it means the CashNote was not spent yet. This is not an error in our DAG building, it simply indicates there is no need to fetch further as unspent money has no descendants.